YoutubeKit/API/ApiSession.swift

```diff
@@ -21,31 +21,83 @@ public class ApiSession {
             
             // If the dataTask error is occured.
             if let error = error {
-                closure(.failed(ResponseError.unexpectedResponse(error)))
+                DispatchQueue.main.async {
```

Comment:
Why do you add `DispatchQueue.main.async`?

Reply:
I added `DispatchQueue.main.async` because I heard that It is a good practice to make the `completion block` (closure) of an async function to run on the `main queue` so that the caller will not get confused  easily. For example, in `Alamofire`, they make their `completionHandler` run on the `main queue` by default, but the caller can specify their own if they want. See this code snippet in line 106 of `ResponseSerialization.swift` in `Alamofire`'s source: 
```
extension DataRequest {
    /// Adds a handler to be called once the request has finished.
    ///
    /// - parameter queue:             The queue on which the completion handler is dispatched.
    /// - parameter completionHandler: The code to be executed once the request has finished.
    ///
    /// - returns: The request.
    @discardableResult
    public func response(queue: DispatchQueue? = nil, completionHandler: @escaping (DefaultDataResponse) -> Void) -> Self {
        delegate.queue.addOperation {
            (queue ?? DispatchQueue.main).async {
                var dataResponse = DefaultDataResponse(
                    request: self.request,
                    response: self.response,
                    data: self.delegate.data,
                    error: self.delegate.error,
                    timeline: self.timeline
                )

                dataResponse.add(self.delegate.metrics)

                completionHandler(dataResponse)
            }
        }

        return self
    }
```
Please let me know your experience and opinion, I am happy to learn. 😄